src/main/java/redis/clients/jedis/JedisClusterInfoCache.java

```diff
@@ -104,12 +104,10 @@ public JedisClusterInfoCache(final GenericObjectPoolConfig<Jedis> poolConfig,
   }
 
   public void discoverClusterNodesAndSlots(Jedis jedis) {
+    List<Object> slots = jedis.clusterSlots();
     w.lock();
-
     try {
       reset();
-      List<Object> slots = jedis.clusterSlots();
-
```

Comment:
@yangbodong22011 @adiamzn If you want this change to be similar to `discoverClusterSlots` which is a private method and all the calling places are within try-catch block. So there is safeguard for `clusterSlots()`. But this `discoverClusterNodesAndSlots` method is a public one and this change would remove existing safeguard.

Reply:
@sazzad16 @yangbodong22011 both options seem fine with me. The critical part here is the lost data in case of failure. 
@sazzad16 From what I see here the try catch block is just for the lock, so we release it even if there is an exception.
I don't mind put the `clusterSlots()` inside the try catch block.
Please let me know what you prefer and I will make the change

Reply:
@adiamzn Ohh, thanks. And, sorry, I missed there isn't any `catch`.